diff --git a/CMakeLists.txt b/CMakeLists.txt
index fd4a1eade..24e08eb2c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -34,6 +34,7 @@ option(protobuf_BUILD_CONFORMANCE "Build conformance tests" OFF)
 option(protobuf_BUILD_EXAMPLES "Build examples" OFF)
 option(protobuf_BUILD_PROTOBUF_BINARIES "Build protobuf libraries and protoc compiler" ON)
 option(protobuf_BUILD_PROTOC_BINARIES "Build libprotoc and protoc compiler" ON)
+option(protobuf_BUILD_PYEXT "Build pyext" ON)
 option(protobuf_BUILD_LIBPROTOC "Build libprotoc" OFF)
 option(protobuf_BUILD_LIBUPB "Build libupb" ON)
 option(protobuf_DISABLE_RTTI "Remove runtime type information in the binaries" OFF)
@@ -338,6 +339,10 @@ if (protobuf_BUILD_CONFORMANCE)
   include(${protobuf_SOURCE_DIR}/cmake/conformance.cmake)
 endif (protobuf_BUILD_CONFORMANCE)
 
+if(protobuf_BUILD_PYEXT)
+  include(${protobuf_SOURCE_DIR}/cmake/pyext.cmake)
+endif(protobuf_BUILD_PYEXT)
+
 if (protobuf_INSTALL)
   include(${protobuf_SOURCE_DIR}/cmake/install.cmake)
 endif (protobuf_INSTALL)
diff --git a/cmake/install.cmake b/cmake/install.cmake
index 076d45f5e..19c64a562 100644
--- a/cmake/install.cmake
+++ b/cmake/install.cmake
@@ -37,6 +37,9 @@ set(_protobuf_libraries libprotobuf-lite libprotobuf)
 if (protobuf_BUILD_LIBPROTOC)
     list(APPEND _protobuf_libraries libprotoc)
 endif (protobuf_BUILD_LIBPROTOC)
+if(protobuf_BUILD_PYEXT)
+  list(APPEND _protobuf_libraries pyext)
+endif(protobuf_BUILD_PYEXT)
 if (protobuf_BUILD_LIBUPB)
   list(APPEND _protobuf_libraries libupb)
 endif ()
@@ -52,9 +55,19 @@ foreach(_library ${_protobuf_libraries})
   install(TARGETS ${_library} EXPORT protobuf-targets
     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT ${_library}
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT ${_library}
-    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT ${_library})
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT ${_library}
+    PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} COMPONENT ${_library})
 endforeach()
 
+# Install pyext headers
+if(protobuf_BUILD_PYEXT)
+  foreach(pyext_header ${pyext_headers})
+    file(RELATIVE_PATH relative_subdir ${pyext_dir} ${pyext_header})
+    get_filename_component(subdir ${relative_subdir} DIRECTORY)
+    install(FILES ${pyext_header} DESTINATION "include/google/protobuf/pyext/${subdir}")
+  endforeach()
+endif(protobuf_BUILD_PYEXT)
+
 if (protobuf_BUILD_PROTOC_BINARIES)
   set(_protobuf_binaries protoc)
   install(TARGETS protoc EXPORT protobuf-targets
diff --git a/cmake/pyext.cmake b/cmake/pyext.cmake
new file mode 100644
index 000000000..281919d22
--- /dev/null
+++ b/cmake/pyext.cmake
@@ -0,0 +1,30 @@
+cmake_policy(SET CMP0057 NEW)
+cmake_policy(SET CMP0074 NEW)
+
+find_package(Python3 COMPONENTS Development REQUIRED)
+
+set(pyext_dir ${CMAKE_CURRENT_SOURCE_DIR}/python/google/protobuf/pyext)
+message("pyext_dir ${pyext_dir}")
+file(GLOB_RECURSE pyext_files
+  LIST_DIRECTORIES false
+  ${pyext_dir}/*.cc
+  ${pyext_dir}/*.h
+)
+
+file(GLOB_RECURSE pyext_headers
+  LIST_DIRECTORIES false
+  ${pyext_dir}/*.h
+)
+
+if(MSVC)
+    # Ignore const string conversion and integer precision loss
+    add_compile_options(/Zc:strictStrings- /wd4244 /wd4267)
+endif()
+
+add_library(pyext STATIC ${pyext_files})
+target_include_directories(pyext PUBLIC
+  $<BUILD_INTERFACE:${protobuf_SOURCE_DIR}/python>
+  $<BUILD_INTERFACE:${pyext_dir}>
+  $<INSTALL_INTERFACE:include>
+)
+target_link_libraries(pyext PUBLIC libprotobuf Python3::Python)
\ No newline at end of file
