diff --git a/src/greenlet/PyGreenlet.cpp b/src/greenlet/PyGreenlet.cpp
index 29c0bba..3c7c24f 100644
--- a/src/greenlet/PyGreenlet.cpp
+++ b/src/greenlet/PyGreenlet.cpp
@@ -667,68 +667,97 @@ green_repr(PyGreenlet* _self)
 
 static PyMethodDef green_methods[] = {
     {
-      .ml_name="switch",
-      .ml_meth=reinterpret_cast<PyCFunction>(green_switch),
-      .ml_flags=METH_VARARGS | METH_KEYWORDS,
-      .ml_doc=green_switch_doc
+      "switch",
+      reinterpret_cast<PyCFunction>(green_switch),
+      METH_VARARGS | METH_KEYWORDS,
+      green_switch_doc
     },
-    {.ml_name="throw", .ml_meth=(PyCFunction)green_throw, .ml_flags=METH_VARARGS, .ml_doc=green_throw_doc},
-    {.ml_name="__getstate__", .ml_meth=(PyCFunction)green_getstate, .ml_flags=METH_NOARGS, .ml_doc=NULL},
-    {.ml_name=NULL, .ml_meth=NULL} /* sentinel */
+    {"throw", (PyCFunction)green_throw, METH_VARARGS, green_throw_doc},
+    {"__getstate__", (PyCFunction)green_getstate, METH_NOARGS, NULL},
+    {NULL, NULL} /* sentinel */
 };
 
 static PyGetSetDef green_getsets[] = {
     /* name, getter, setter, doc, context pointer */
-    {.name="__dict__", .get=(getter)green_getdict, .set=(setter)green_setdict},
-    {.name="run", .get=(getter)green_getrun, .set=(setter)green_setrun},
-    {.name="parent", .get=(getter)green_getparent, .set=(setter)green_setparent},
-    {.name="gr_frame", .get=(getter)green_getframe },
+    {"__dict__", (getter)green_getdict, (setter)green_setdict},
+    {"run", (getter)green_getrun, (setter)green_setrun},
+    {"parent", (getter)green_getparent, (setter)green_setparent},
+    {"gr_frame", (getter)green_getframe },
     {
-      .name="gr_context",
-      .get=(getter)green_getcontext,
-      .set=(setter)green_setcontext
+      "gr_context",
+      (getter)green_getcontext,
+      (setter)green_setcontext
     },
-    {.name="dead", .get=(getter)green_getdead},
-    {.name="_stack_saved", .get=(getter)green_get_stack_saved},
-    {.name=NULL}
+    {"dead", (getter)green_getdead},
+    {"_stack_saved", (getter)green_get_stack_saved},
+    {NULL}
 };
 
 static PyMemberDef green_members[] = {
-    {.name=NULL}
+    {NULL}
 };
 
 static PyNumberMethods green_as_number = {
-  .nb_bool=(inquiry)green_bool,
+  0, /* nb_add */
+  0, /* nb_subtract */
+  0, /* nb_multiply */
+  0, /* nb_remainder */
+  0, /* nb_divmod */
+  0, /* nb_power */
+  0, /* nb_negative */
+  0, /* nb_positive */
+  0, /* nb_absolute */
+  (inquiry)green_bool,
 };
 
 
 PyTypeObject PyGreenlet_Type = {
-    .ob_base=PyVarObject_HEAD_INIT(NULL, 0)
-    .tp_name="greenlet.greenlet", /* tp_name */
-    .tp_basicsize=sizeof(PyGreenlet),  /* tp_basicsize */
+    PyVarObject_HEAD_INIT(NULL, 0)
+    "greenlet.greenlet", /* tp_name */
+    sizeof(PyGreenlet),  /* tp_basicsize */
     /* methods */
-    .tp_dealloc=(destructor)green_dealloc, /* tp_dealloc */
-    .tp_repr=(reprfunc)green_repr,      /* tp_repr */
-    .tp_as_number=&green_as_number,          /* tp_as _number*/
-    .tp_flags=G_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE, /* tp_flags */
-    .tp_doc="greenlet(run=None, parent=None) -> greenlet\n\n"
+    0, /* tp_itemsize */
+    (destructor)green_dealloc, /* tp_dealloc */
+    0, /* tp_vectorcall_offset */
+    0, /* tp_getattr */
+    0, /* tp_setattr */
+    0, /* tp_as_async */
+    (reprfunc)green_repr,      /* tp_repr */
+    &green_as_number,          /* tp_as _number*/
+    0, /* tp_as_sequence */
+    0, /* tp_as_mapping */
+    0, /* tp_hash */
+    0, /* tp_call */
+    0, /* tp_str */
+    0, /* tp_getattro */
+    0, /* tp_setattro */
+    0, /* tp_as_buffer */
+    G_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE, /* tp_flags */
+    "greenlet(run=None, parent=None) -> greenlet\n\n"
     "Creates a new greenlet object (without running it).\n\n"
     " - *run* -- The callable to invoke.\n"
     " - *parent* -- The parent greenlet. The default is the current "
     "greenlet.",                        /* tp_doc */
-    .tp_traverse=(traverseproc)green_traverse, /* tp_traverse */
-    .tp_clear=(inquiry)green_clear,         /* tp_clear */
-    .tp_weaklistoffset=offsetof(PyGreenlet, weakreflist),  /* tp_weaklistoffset */
-
-    .tp_methods=green_methods,                      /* tp_methods */
-    .tp_members=green_members,                      /* tp_members */
-    .tp_getset=green_getsets,                      /* tp_getset */
-    .tp_dictoffset=offsetof(PyGreenlet, dict),         /* tp_dictoffset */
-    .tp_init=(initproc)green_init,               /* tp_init */
-    .tp_alloc=PyType_GenericAlloc,                  /* tp_alloc */
-    .tp_new=(newfunc)green_new,                          /* tp_new */
-    .tp_free=PyObject_GC_Del,                   /* tp_free */
-    .tp_is_gc=(inquiry)green_is_gc,         /* tp_is_gc */
+    (traverseproc)green_traverse, /* tp_traverse */
+    (inquiry)green_clear,         /* tp_clear */
+    0, /* tp_richcompare */
+    offsetof(PyGreenlet, weakreflist),  /* tp_weaklistoffset */
+
+    0, /* tp_iter */
+    0, /* tp_iternext */
+    green_methods,                      /* tp_methods */
+    green_members,                      /* tp_members */
+    green_getsets,                      /* tp_getset */
+    0, /* tp_base */
+    0, /* tp_dict */
+    0, /* tp_descr_get */
+    0, /* tp_descr_set */
+    offsetof(PyGreenlet, dict),         /* tp_dictoffset */
+    (initproc)green_init,               /* tp_init */
+    PyType_GenericAlloc,                  /* tp_alloc */
+    (newfunc)green_new,                          /* tp_new */
+    PyObject_GC_Del,                   /* tp_free */
+    (inquiry)green_is_gc,         /* tp_is_gc */
 };
 
 #endif
diff --git a/src/greenlet/PyGreenletUnswitchable.cpp b/src/greenlet/PyGreenletUnswitchable.cpp
index 1b768ee..f685320 100644
--- a/src/greenlet/PyGreenletUnswitchable.cpp
+++ b/src/greenlet/PyGreenletUnswitchable.cpp
@@ -111,36 +111,62 @@ green_unswitchable_setforceslp(PyGreenlet* self, PyObject* nforce, void* UNUSED(
 static PyGetSetDef green_unswitchable_getsets[] = {
     /* name, getter, setter, doc, closure (context pointer) */
     {
-      .name="force_switch_error",
-      .get=(getter)green_unswitchable_getforce,
-      .set=(setter)green_unswitchable_setforce,
-      .doc=NULL
+      "force_switch_error",
+      (getter)green_unswitchable_getforce,
+      (setter)green_unswitchable_setforce,
+      NULL
     },
     {
-      .name="force_slp_switch_error",
-      .get=(getter)green_unswitchable_getforceslp,
-      .set=(setter)green_unswitchable_setforceslp,
-      .doc=nullptr
+      "force_slp_switch_error",
+      (getter)green_unswitchable_getforceslp,
+      (setter)green_unswitchable_setforceslp,
+      nullptr
     },
-    {.name=nullptr}
+    {nullptr}
 };
 
 PyTypeObject PyGreenletUnswitchable_Type = {
-    .ob_base=PyVarObject_HEAD_INIT(NULL, 0)
-    .tp_name="greenlet._greenlet.UnswitchableGreenlet",
-    .tp_dealloc= (destructor)green_dealloc, /* tp_dealloc */
-    .tp_flags=G_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE, /* tp_flags */
-    .tp_doc="Undocumented internal class",                        /* tp_doc */
-    .tp_traverse=(traverseproc)green_traverse, /* tp_traverse */
-    .tp_clear=(inquiry)green_clear,         /* tp_clear */
-
-    .tp_getset=green_unswitchable_getsets,                      /* tp_getset */
-    .tp_base=&PyGreenlet_Type,                                  /* tp_base */
-    .tp_init=(initproc)green_init,               /* tp_init */
-    .tp_alloc=PyType_GenericAlloc,                  /* tp_alloc */
-    .tp_new=(newfunc)green_unswitchable_new,                          /* tp_new */
-    .tp_free=PyObject_GC_Del,                   /* tp_free */
-    .tp_is_gc=(inquiry)green_is_gc,         /* tp_is_gc */
+    PyVarObject_HEAD_INIT(NULL, 0)
+    "greenlet._greenlet.UnswitchableGreenlet",
+    0, /* tp_basicsize */
+    0, /* tp_itemsize */
+    (destructor)green_dealloc, /* tp_dealloc */
+    0, /* tp_vectorcall_offset */
+    0, /* tp_getattr */
+    0, /* tp_setattr */
+    0, /* tp_as_async */
+    0, /* tp_repr */
+    0, /* tp_as_number */
+    0, /* tp_as_sequence */
+    0, /* tp_as_mapping */
+    0, /* tp_hash */
+    0, /* tp_call */
+    0, /* tp_str */
+    0, /* tp_getattro */
+    0, /* tp_setattro */
+    0, /* tp_as_buffer */
+    G_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE, /* tp_flags */
+    "Undocumented internal class",                        /* tp_doc */
+    (traverseproc)green_traverse, /* tp_traverse */
+    (inquiry)green_clear,         /* tp_clear */
+
+    0, /* tp_richcompare */
+    0, /* tp_weaklistoffset */
+    0, /* tp_iter */
+    0, /* tp_iternext */
+    0, /* tp_methods */
+    0, /* tp_members */
+    green_unswitchable_getsets,                      /* tp_getset */
+    &PyGreenlet_Type,                                  /* tp_base */
+    0, /* tp_dict */
+    0, /* tp_descr_get */
+    0, /* tp_descr_set */
+    0, /* tp_dictoffset */
+    (initproc)green_init,               /* tp_init */
+    PyType_GenericAlloc,                  /* tp_alloc */
+    (newfunc)green_unswitchable_new,                          /* tp_new */
+    PyObject_GC_Del,                   /* tp_free */
+    (inquiry)green_is_gc,         /* tp_is_gc */
 };
 
 
diff --git a/src/greenlet/PyModule.cpp b/src/greenlet/PyModule.cpp
index 6adcb5c..0554226 100644
--- a/src/greenlet/PyModule.cpp
+++ b/src/greenlet/PyModule.cpp
@@ -207,62 +207,62 @@ mod_get_tstate_trash_delete_nesting(PyObject* UNUSED(module))
 
 static PyMethodDef GreenMethods[] = {
     {
-      .ml_name="getcurrent",
-      .ml_meth=(PyCFunction)mod_getcurrent,
-      .ml_flags=METH_NOARGS,
-      .ml_doc=mod_getcurrent_doc
+      "getcurrent",
+      (PyCFunction)mod_getcurrent,
+      METH_NOARGS,
+      mod_getcurrent_doc
     },
     {
-      .ml_name="settrace",
-      .ml_meth=(PyCFunction)mod_settrace,
-      .ml_flags=METH_VARARGS,
-      .ml_doc=mod_settrace_doc
+      "settrace",
+      (PyCFunction)mod_settrace,
+      METH_VARARGS,
+      mod_settrace_doc
     },
     {
-      .ml_name="gettrace",
-      .ml_meth=(PyCFunction)mod_gettrace,
-      .ml_flags=METH_NOARGS,
-      .ml_doc=mod_gettrace_doc
+      "gettrace",
+      (PyCFunction)mod_gettrace,
+      METH_NOARGS,
+      mod_gettrace_doc
     },
     {
-      .ml_name="set_thread_local",
-      .ml_meth=(PyCFunction)mod_set_thread_local,
-      .ml_flags=METH_VARARGS,
-      .ml_doc=mod_set_thread_local_doc
+      "set_thread_local",
+      (PyCFunction)mod_set_thread_local,
+      METH_VARARGS,
+      mod_set_thread_local_doc
     },
     {
-      .ml_name="get_pending_cleanup_count",
-      .ml_meth=(PyCFunction)mod_get_pending_cleanup_count,
-      .ml_flags=METH_NOARGS,
-      .ml_doc=mod_get_pending_cleanup_count_doc
+      "get_pending_cleanup_count",
+      (PyCFunction)mod_get_pending_cleanup_count,
+      METH_NOARGS,
+      mod_get_pending_cleanup_count_doc
     },
     {
-      .ml_name="get_total_main_greenlets",
-      .ml_meth=(PyCFunction)mod_get_total_main_greenlets,
-      .ml_flags=METH_NOARGS,
-      .ml_doc=mod_get_total_main_greenlets_doc
+      "get_total_main_greenlets",
+      (PyCFunction)mod_get_total_main_greenlets,
+      METH_NOARGS,
+      mod_get_total_main_greenlets_doc
     },
     {
-      .ml_name="get_clocks_used_doing_optional_cleanup",
-      .ml_meth=(PyCFunction)mod_get_clocks_used_doing_optional_cleanup,
-      .ml_flags=METH_NOARGS,
-      .ml_doc=mod_get_clocks_used_doing_optional_cleanup_doc
+      "get_clocks_used_doing_optional_cleanup",
+      (PyCFunction)mod_get_clocks_used_doing_optional_cleanup,
+      METH_NOARGS,
+      mod_get_clocks_used_doing_optional_cleanup_doc
     },
     {
-      .ml_name="enable_optional_cleanup",
-      .ml_meth=(PyCFunction)mod_enable_optional_cleanup,
-      .ml_flags=METH_O,
-      .ml_doc=mod_enable_optional_cleanup_doc
+      "enable_optional_cleanup",
+      (PyCFunction)mod_enable_optional_cleanup,
+      METH_O,
+      mod_enable_optional_cleanup_doc
     },
 #if !GREENLET_PY313
     {
-      .ml_name="get_tstate_trash_delete_nesting",
-      .ml_meth=(PyCFunction)mod_get_tstate_trash_delete_nesting,
-      .ml_flags=METH_NOARGS,
-      .ml_doc=mod_get_tstate_trash_delete_nesting_doc
+      "get_tstate_trash_delete_nesting",
+      (PyCFunction)mod_get_tstate_trash_delete_nesting,
+      METH_NOARGS,
+      mod_get_tstate_trash_delete_nesting_doc
     },
 #endif
-    {.ml_name=NULL, .ml_meth=NULL} /* Sentinel */
+    {NULL, NULL} /* Sentinel */
 };
 
 static const char* const copy_on_greentype[] = {
@@ -275,11 +275,11 @@ static const char* const copy_on_greentype[] = {
 };
 
 static struct PyModuleDef greenlet_module_def = {
-    .m_base=PyModuleDef_HEAD_INIT,
-    .m_name="greenlet._greenlet",
-    .m_doc=NULL,
-    .m_size=-1,
-    .m_methods=GreenMethods,
+    PyModuleDef_HEAD_INIT,
+    "greenlet._greenlet",
+    NULL,
+    -1,
+    GreenMethods,
 };
 
 
